SolutionsAndTests

```diff
@@ -0,0 +1,258 @@
+		return false;
+	}
+	return true;
+}
+
+function testProblem2()
+{
+	var simpleArray1 = [1, 2, 3, 4]
+	var simpleArray1Response = problem2(simpleArray1);
+	if(!(Object.prototype.toString.call( simpleArray1Response ) === '[object Array]' ))
+	{
+		//alert("simpleArray1=>" + simpleArray1Response);
+		return false;
+	}
+	if (simpleArray1Response.length != 2)
+	{
+		//alert("simpleArray1=>" + simpleArray1Response);
+		return false;
+	}
+	if (simpleArray1Response[0] != 4 || simpleArray1Response[1] != 3)
+	{
+		//alert("simpleArray1=>" + simpleArray1Response);
+		return false;
+	}
+	
+	var simpleArray2 = [4, 3, 2, 1]
+	var simpleArray2Response = problem2(simpleArray2);
+	if(!(Object.prototype.toString.call( simpleArray2Response ) === '[object Array]' ))
+	{
+		//alert("simpleArray1=>" + simpleArray2Response);
+		return false;
+	}
+	if (simpleArray2Response.length != 2)
+	{
+		//alert("simpleArray1=>" + simpleArray2Response);
+		return false;
+	}
+	if (simpleArray2Response[0] != 4 || simpleArray2Response[1] != 3)
+	{
+		//alert("simpleArray1=>" + simpleArray2Response);
+		return false;
+	}
+	
+	var oneElementArray = [1]
+	var oneElementArrayResponse = problem2(oneElementArray);
+	if(!(Object.prototype.toString.call( oneElementArrayResponse ) === '[object Array]' ))
+	{
+		//alert("simpleArray1=>" + oneElementArrayResponse);
+		return false;
+	}
+	if (oneElementArrayResponse.length != 1)
+	{
+		//alert("simpleArray1=>" + oneElementArrayResponse);
+		return false;
+	}
+	if (oneElementArrayResponse[0] != 1)
+	{
+		//alert("simpleArray1=>" + oneElementArrayResponse);
+		return false;
+	}
+	return true;
+}
+
+
+function testProblem3()
+{
+	var simpleArray1 = [1, 2, 3, 4]
+	var simpleArray1Response = problem3(simpleArray1);
+	if(simpleArray1Response != false)
+	{
+		//alert("simpleArray1=>" + simpleArray1Response);
+		return false;
+	}
+	
+	var simpleArray2 = [4, 3, 2, 1]
+	var simpleArray2Response = problem3(simpleArray2);
+	if(simpleArray2Response != false)
+	{
+		//alert("simpleArray1=>" + simpleArray2Response);
+		return false;
+	}
+	
+	var duplicatesArray1 = [1, 2, 3, 1]
+	var duplicatesArray1Response = problem3(duplicatesArray1);
+	if(duplicatesArray1Response != true)
+	{
+		//alert("simpleArray1=>" + duplicatesArray1Response);
+		return false;
+	}
+	
+	var duplicatesArray2 = [4, 2, 3, 4]
+	var duplicatesArray2Response = problem3(duplicatesArray2);
+	if(duplicatesArray2Response != true)
+	{
+		//alert("simpleArray1=>" + duplicatesArray2Response);
+		return false;
+	}
+	
+	var duplicatesArray3 = [2, 1, 2, 4]
+	var duplicatesArray3Response = problem3(duplicatesArray3);
+	if(duplicatesArray3Response != true)
+	{
+		//alert("simpleArray1=>" + duplicatesArray3Response);
+		return false;
+	}
+	
+	var duplicatesArray4 = [1, 2, 3, 2]
+	var duplicatesArray4Response = problem3(duplicatesArray4);
+	if(duplicatesArray4Response != true)
+	{
+		//alert("simpleArray1=>" + duplicatesArray4Response);
+		return false;
+	}
+	return true;
+}
```

Comment:
These are not proper unit tests. A nice framework for doing unit tests in javascript is [jasmine](http://jasmine.github.io/2.3/introduction.html).